msbuild/rest-api-client-generation/README.md

```diff
@@ -0,0 +1,272 @@
+
+- At this point in time we can execute a Visual Studio Solution rebuild and see the client generated on the PetShopRestClient folder.
+- We are going to use the generated client. Go to the client Program.cs and copy the following code
+
+  ```csharp
+  using System;
+  using System.Net.Http;
+
+  namespace PetReaderExecTaskExample
+  {
+      internal class Program
+      {
+          private const string baseUrl = "https://petstore.swagger.io/v2";
+          static void Main(string[] args)
+          {
+              HttpClient httpClient = new HttpClient();
+              httpClient.BaseAddress = new Uri(baseUrl);
+              var petClient = new PetShopRestClient.PetShopRestClient(httpClient);
+              var pet = petClient.GetPetByIdAsync(1).Result;
+              Console.WriteLine($"Id: {pet.Id} Name: {pet.Name} Status: {pet.Status} CategoryName: {pet.Category.Name}");
+          }
+      }
+  }
+  ```
+
+  _Note:_ We are using `new HttpClient()` because is simple to show our example working, but it is not appropriated. The best practice is to use HTTPClientFactory to create an HTTPClient object which addresses the known issues of HTTPClient request like Resource Exhaustion or Stale DNS problems. [Read more about it](https://docs.microsoft.com/dotnet/architecture/microservices/implement-resilient-applications/use-httpclientfactory-to-implement-resilient-http-requests)
+
+- Congrats!! Now, you can execute the program to see how it is working
+
+## Option 2: Use the Custom task derived from MSBuid Tool Task
+
+In many cases the option 1 is good enough to execute an external tools to do something, like Rest Api Client Code Generation.
+We are going to continue with the same example, but the ideas can be used for others examples.
+What if we want to allow Rest Api Client Code Generation if only if we don't use absolute Windows path as input? Or What if we need to calculate in some way where the executable is?.
+When there is any situation where we need execute some code to do extra work, the [MSBuild Tool Task](https://docs.microsoft.com/dotnet/api/microsoft.build.utilities.tooltask) is the best solution. This is an abstract class derivated from MSBuild Task, we need to define a concrete subclass (_We will need to create a Custom MSBuild Task_). It is prepare for command execution and allows us to introduce code during the process.
+
+We are going to generate a Custom Task derivated from [MSBuild Tool Task](https://docs.microsoft.com/dotnet/api/microsoft.build.utilities.tooltask) which will generate a Rest API client but it will fail if we try to reference the OpenApi spec using a http address. NSwag supports a http address as OpenApi spec input, but we don't want to allow that. It is recommended read about [Custom task code generation](../custom-task-code-generation/) first, that article contains useful information about Custom Task creation.
```

Comment:
```suggestion
We are going to generate a Custom Task derived from [MSBuild Tool Task](https://docs.microsoft.com/dotnet/api/microsoft.build.utilities.tooltask) which will generate a Rest API client but it will fail if we try to reference the OpenApi spec using a http address. NSwag supports a http address as OpenApi spec input, but we don't want to allow that. It is recommended read about [Custom task code generation](../custom-task-code-generation/) first, that article contains useful information about Custom Task creation.
```